server/store.go

```diff
@@ -155,6 +157,16 @@ func (s *StoreStatus) GetUptime() time.Duration {
 	return 0
 }
 
+// GetHeartbeatInterval roughly calculates the heartbeat interval.
+// returns zero interval when there is no heartbeat arrived.
+func (s *StoreStatus) GetHeartbeatInterval() time.Duration {
+	if s.HeartbeatCount == 0 {
+		return time.Duration(0)
+	}
+
+	return time.Duration(int64(s.LastHeartbeatTS.Sub(s.StartTS)) / s.HeartbeatCount)
```

Comment:
This doesn't look reliable. How about sending interval along with Heartbeat request from tikv-server?